### DISCUSSION TITLE
SEO Improvements using NextJS

### DISCUSSION THREAD

**Discussion by rperozin** [scandipwa/scandipwa#3387] [Q&A] [answered]
I'm setting up the ScandiPWA for my project and would like to use [NextJS](https://nextjs.org/) to make it **server-side-rendering** to improve the **SEO performance**.

Anyone already did it?
Has any option related to this?

I'm not finding anything.

**Comment by Rijaspb** [comment]
https://www.rijasexports.com/

**Comment by rperozin** [comment, marked as answer]
I spoke by email with ScandiPWA support, and they told me the NextJS is on their radar, but it is a feature on the list only for the next year. To use it on my project, I need to refactor it by myself. But they suggested using the [Prerender.io](https://prerender.io/) with [ReadyMage](https://help.readymage.com/service-management/prerender) extension as an alternative for the server-side-rendering.

The **Prerender** middleware will check each request to see if it’s a request from a crawler. If it is a request from a crawler, the middleware will send a request to Prerender.io for the static HTML of that page. If not, the request will continue to the regular server routes. The crawler never knows that you are using Prerender.io since the response always goes through your server.

Cons of using Prerender instead of NextJS:
- It has a cost
- It’s not NextJS
- It requires using the ReadyMage extension for it, and this also has a cost.
- It’s solving the Google Robot issues, but it will not improve the user experience (loading time).

**Comment by biuro-mirat** [comment]
Can SEO problems be expected without NextJS?

**Reply by rperozin** [reply]
I don't think so. If you are doing the SSR (Server-side-render) for the Google robot all should be solved.

But if you don't do the SSR, you could have these as consequences:
- Slow and complex indexing process.
- Any error in JavaScript code could make it impossible to index.
- It could [exhaust the crawling budget](https://rubygarage.org/blog/seo-for-react-websites#article_title_3).
- SPAs have dynamic content, and it could make the bot find an empty page.

From my research, the main difference between NextJS and Prerender.io is for the user that could need to wait a bit more until the page fully loads. And for you, as a developer, you will need to find another way to do the URL redirections.

**Reply by rperozin** [reply]
But SEO is not only this. You also need to be aware of other points like using the proper element tags, not overload the page with keywords, friendly URL, naming correctly the images, use meta tags correctly and a lot of other strategies that you can ask some SEO specialist.

**Comment by biuro-mirat** [comment]
And what does it look like when rendering on the user side?

**Reply by rperozin** [reply]
I suggest you to read this article about SSR and CSR
https://rubygarage.org/blog/seo-for-react-websites#article_title_1

**Comment by biuro-mirat** [comment]
Thank you for the information.
